File: src/event_schemas/user/events.py
from pydantic import BaseModel


class UserCreated(BaseModel):
    id: int
    name: str


class UserUpdated(BaseModel):
    id: int
    name: str


registry: dict[str, type[BaseModel]] = {
    'user_created': UserCreated,
    'user_updated': UserUpdated,
}

File: src/schema/order/events.go
// # from pydantic import BaseModel

// # class OrderCreated(BaseModel):
// #     id: int
// #     user_id: int
// #     product_id: int
// #     amount: int

// # registry: dict[str, type[BaseModel]] = {
// #     'order_created': OrderCreated,
// # }

package order

type OrderCreated struct {
	Id        int `json:"id" validate:"required"`
	UserId    int `json:"user_id" validate:"required"`
	ProductId int `json:"product_id" validate:"required"`
	Amount    int `json:"amount" validate:"required"`
}
func (event *OrderCreated) EventType () string{
	return "order_created"
}

